src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -1652,5 +1652,110 @@ public void ChainWith10DataAvailableHeaders_PartialValidationSucceededOnTheFirst
             listOfHeaders.Should().HaveCount(1);
             listOfHeaders.First().ChainedHeader.HashBlock.Should().Be(secondHeader.HashBlock);
         }
+
+        /// <summary>
+        /// Issue 51 @ Call PartialValidationSucceeded on a chained header which doesn't have the block data but it is
+        /// in the tree, make sure PartialValidationSucceeded returns null and ReorgRequired is false.
+        /// </summary>
+        [Fact]
+        public void CallingPartialValidationSucceeded_ForConnectedHeaderWithNoBlockData_NothingReturnedAndFullValidationIsNotRequired()
+        {
+            // Chain header tree setup. Initial chain has 5 headers with no blocks.
+            // Example: h1=h2=h3=h4=h5.
+            const int initialChainSize = 5;
+            TestContext ctx = new TestContextBuilder()
+                .WithInitialChain(initialChainSize, assignBlocks: false)
+                .Build();
+            ChainedHeaderTree cht = ctx.ChainedHeaderTree;
+            ChainedHeader chainTip = ctx.InitialChainTip;
+            
+            // Call partial validation on h4 and make sure nothing is returned
+            // and full validation is not required.
+            List<ChainedHeaderBlock> headers = cht.PartialValidationSucceeded(chainTip, out bool fullValidationRequired);
```

Comment:
Actually, this test is also not right. We need to present the headers and then call this one of the headers that there consumed, because the initial chain is fully validated by definition already.